src/sys/unix/uds/datagram.rs

```diff
@@ -163,7 +164,23 @@ impl FromRawFd for UnixDatagram {
 }
 
 impl IntoRawFd for UnixDatagram {
-    fn into_raw_fd(self) -> RawFd {
-        self.inner.into_raw_fd()
+    fn into_raw_fd(mut self) -> RawFd {
+        let fd = mem::replace(&mut self.inner, unsafe {
+            mem::MaybeUninit::uninit().assume_init()
```

Comment:
We could use`net::UnixDatagram::from_raw_fd(-1)`, then at least we don't have any memory unsafety.